### PR TITLE
Learning merge (Nuve/Mongodb removal)

### PR DESCRIPTION
**Pull request by caboteria**
This is based on the learning branch, but I re-arranged the commits so they each stand alone better.  I added one commit that removes most of Nuve, except the tests.  We'll have to go back and fix the tests so they don't need Nuve.

It's probably easier to look at this PR in terms of its commits rather than the total file changes.

### CODE REVIEW

**Comment by j4y** [comment]
:+1:

File: erizo_controller/erizoController/erizoController.js [hunk 1]
```diff
@@ -199,96 +178,13 @@ var addToCloudHandler = function (callback) {
...
-            }
-            if (msg == 'error') {
-                log.info('Error in communication with cloudProvider');
-            }
-
-            publicIP = msg.publicIP;
-            myId = msg.id;
-            myState = 2;
+    log.debug('publicIp:', publicIP);
+    myId = 42;   // FIXME: does this need to vary or can it stay hard-coded?
```
Comment (MorikoHandford) [inline erizo_controller/erizoController/erizoController.js:182]: Is this something to look at in the future?

File: extras/basic_example/public/connection_test.js [hunk 2]
```diff
@@ -1,12 +1,6 @@
...
-var localStream, room;
-
-function getParameterByName(name) {
-  name = name.replace(/[\[]/, "\\\[").replace(/[\]]/, "\\\]");
-  var regex = new RegExp("[\\?&]" + name + "=([^&#]*)"),
-      results = regex.exec(location.search);
-  return results == null ? "" : decodeURIComponent(results[1].replace(/\+/g, " "));
-}
+var localStream, room,
+    sessionId = Math.random() * 1000;  // FIXME: get sessionId from somewhere
```
Comment (MorikoHandford) [inline extras/basic_example/public/connection_test.js:3]: Is this and the other FIXMEs for future releases?

File: extras/basic_example/public/connection_test.js [hunk 3]
```diff
@@ -15,76 +9,64 @@ function printText(text) {
...
-    var req = new XMLHttpRequest();
-    var url = serverUrl + 'createToken/';
-    var body = {username: userName, role: role};
-
-    req.onreadystatechange = function () {
-      if (req.readyState === 4) {
-        callback(req.responseText);
+  var token = JSON.stringify({
+    "id": "session-" + sessionId,
+    "host": "192.168.33.99:8080",  // FIXME: needs to come from the server
```
Comment (MorikoHandford) [inline extras/basic_example/public/connection_test.js:14]: Doesn't this info come from the config file?  Do we need this code?

File: extras/basic_example/public/connection_test.js [hunk 4]
```diff
@@ -15,76 +9,64 @@ function printText(text) {
...
+    "secure": false
+  });
+  console.log(token);
+  room = Erizo.Room({token: token});
+
+  localStream.addEventListener("access-accepted", function () {
+    printText('Mic and Cam OK');
```
Comment (MorikoHandford) [inline extras/basic_example/public/connection_test.js:21]: Are we recording voice?

**Comment by MorikoHandford** [comment]
Awesome that we got rid of so much stuff!  I'm not sure any of what I committed on needs a change.  Thanks @caboteria .

**Comment by MorikoHandford** [comment]
Is this going to be merged into the Affectiva branch or the master branch?

**Comment by caboteria** [comment]
master

**Pull request merged**